### PR TITLE
Add cni-default-network option

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Supersedes https://github.com/cri-o/cri-o/pull/2123
Fixes https://github.com/cri-o/cri-o/issues/2121

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
- Added `--cni-default-network`/`cni_default_network` option to specify the CNI network
  to select. The default value is `crio`, but this option can be explicitly set to `""` to pickup
  the first network found in `--cni-config-dir`/`network_dir`.
```

### CODE REVIEW

**Comment by haircommander** [comment]
LGTM, thanks for picking this up @saschagrunert

File: pkg/config/config.go [hunk 1]
```diff
@@ -562,8 +565,9 @@ func DefaultConfig() (*Config, error) {
 			ImageVolumes:     ImageVolumesMkdir,
 		},
 		NetworkConfig: NetworkConfig{
-			NetworkDir: cniConfigDir,
-			PluginDirs: []string{cniBinDir},
+			CNIDefaultNetwork: "crio",
```
Comment (mrunalp) [inline pkg/config/config.go]: I think we should default to empty.

Comment (saschagrunert) [reply pkg/config/config.go]: Alright, I changed the implementation and the docs to default to `""`

**Comment by haircommander** [comment]
needs rebase :confused:

**Comment by saschagrunert** [comment]
@haircommander rebased

**Pull request merged**